tests/cupyx_tests/scipy_tests/stats_tests/test_morestats.py

```diff
@@ -0,0 +1,109 @@
+import cupy
+
+from cupy import testing
+
+import cupyx
+import cupyx.scipy.stats  # NOQA
+
+import scipy.stats  # NOQA
+
+
+atol = {
+    'default': 1e-6,
+    cupy.float16: 5e-2,
+    cupy.float64: 1e-14
+}
+rtol = {
+    'default': 1e-6,
+    cupy.float16: 5e-2,
+    cupy.float64: 1e-14
+}
+
+
+atol_low = {
+    'default': 1e-6,
+    cupy.float16: 5e-3,
+    cupy.float32: 1e-5,
+    cupy.float64: 5e-3
```

Comment:
Seems this tolerance is too large for float64, maybe something wrong.

Reply:
I think there is a problem with the first case only. The rest works with `1e-14`.

Reply:
ah, it's because numpy.boxcox_llf returns float64 outputs for lower precision inputs as float16 and float32, so `cupy.float64` key's value in the dict is used for float16 inputs.

Reply:
Plase split `test_array_1dim` to float16 case and others.

```py
    @testing.for_all_dtypes(no_float16=True)
    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=1e-6)
    def test_array_1dim(self, xp, scp, dtype):
        data = testing.shaped_random((2,), xp, dtype=dtype, scale=9)
        import numpy
        xp.log(data)
        lmb = 4.0
        return scp.stats.boxcox_llf(lmb, data)

    @testing.numpy_cupy_allclose(scipy_name='scp', rtol=5e-3)
    def test_array_1dim_float16(self, xp, scp):
        data = testing.shaped_random((2,), xp, dtype=xp.float16, scale=9)
        lmb = 4.0
        return scp.stats.boxcox_llf(lmb, data)
```

Reply:
Indeed this works. Thanks